test/integ/test_connection.py

```diff
@@ -1192,3 +1192,33 @@ def check_packages(message: str, expected_packages: list[str]) -> bool:
     ) as conn, capture_sf_telemetry.patch_connection(conn, False) as telemetry_test:
         conn._log_telemetry_imported_packages()
         assert len(telemetry_test.records) == 0
+
+
+@pytest.mark.skipolddriver
+def test_connection_name_loading(monkeypatch, db_parameters):
+    import tomlkit
+
+    doc = tomlkit.document()
+    default_con = tomlkit.table()
+    doc["default"] = default_con
+    try:
+        # If anything unexpected fails here, don't want to expose password
+        for k, v in db_parameters.items():
+            default_con[k] = v
+        with monkeypatch.context() as m:
+            m.setenv("SF_CONNECTIONS", tomlkit.dumps(doc))
+            with snowflake.connector.connect(connection_name="default") as conn:
+                with conn.cursor() as cur:
+                    assert cur.execute("select 1;").fetchall() == [
+                        (1,),
+                    ]
+    except Exception:
+        pytest.fail("something failed", pytrace=False)
+
+
+@pytest.mark.skipolddriver
+def test_not_found_connection_name():
```

Comment:
maybe more negative tests? e.g., config file is not present, wrong format, etc.

Reply:
I have unit tests for all those. I made sure we have 100% coverage for `snowflake.connector.config_parser`